code/modules/mob/living/carbon/human/life/handle_hypothermia.dm

```diff
@@ -0,0 +1,84 @@
+var/global/list/coldwarning_light = list("It's a wee bit chilly, isn't it?","Your nose feels numb","It's pretty cold!","You should probably wrap up some more.","You can see your breath!","You feel like taking a nap.")
+var/global/list/coldwarning_hard = list("Holy shit, it's freezing cold out here!","You should probably get warmed up quickly!","You can't feel your hands!","You can't feel anything!","You're absolutely exhausted!")
+
+/mob/living/proc/undergoing_hypothermia()
+	if(!istype(src,/mob/living/carbon/human) && !istype(src,/mob/living/carbon/monkey))
+		return NO_HYPOTHERMIA
+	else
+		var/mob/living/carbon/human/H = src
+		if(H.species == "/datum/species/horror")
+			return NO_HYPOTHERMIA
+	if((status_flags & GODMODE) || (flags & INVULNERABLE) || istype(loc, /obj/machinery/atmospherics/unary/cryo_cell))
+		return NO_HYPOTHERMIA
+	var/body_temp_celcius = src.bodytemperature - T0C
+	switch(body_temp_celcius)
+		if(32 to 35)
+			return MILD_HYPOTHERMIA // awake and shivering
+		if(28 to 32)
+			return MODERATE_HYPOTHERMIA // drowsy and not shivering
+		if(24 to 28)
+			return SEVERE_HYPOTHERMIA // unconcious, not shivering
+		if(-T0C to 24)
+			return PROFOUND_HYPOTHERMIA // no vital signs
+	return NO_HYPOTHERMIA
+
+/mob/living/carbon/human/proc/is_vessel_dilated() // finds out if the blood vessel is dilated - ie expanded and more subsceptible to hypothermia.
+	if(!reagents)
+		return 0
+	var/datum/reagents/blood = reagents
+	var/vessel_dilated = 0
+	if(blood.reagent_list)
+		for(var/datum/reagent/chem in blood.reagent_list)
+			if(istype(chem,/datum/reagent/ethanol)) // alcohol is a vasodilator.
+				vessel_dilated = 1
```

Comment:
You can replace the vessel_dilated things here with return 1, and then have the end of the proc return 0